benchmarks/bench_ml.py

```diff
@@ -0,0 +1,422 @@
+
+
+def evaluate_optimizer(base_estimator, model, dataset, n_calls, seed_value):
+    """
+    Evaluates some estimator for the task of optimization of parameters of some
+    model, given limited number of model evaluations.
+    Parentheses on parameters are used to be able to run function with pool.map method.
+
+    :param base_estimator: Estimator to use for optimization.
+    :param model: str, name of the ML model class to be used for parameter tuning
+    :param dataset: str, name of dataset to train ML model on
+    :param n_calls: a budget of evaluations
+    :param seed_value: random seed, used to set the random number generator in numpy
+    :return: a list of paris (p, f(p)), where p is a dictionary of the form "param name":value,
+            and f(p) is performance measure value achieved by the model for configuration p.
+            Such list contains history of execution of optimization.
+    """
+
+
+    # below seed is necessary for processes which fork at the same time
+    # so that random numbers generated in processes are different
+    np.random.seed(seed_value)
+
+    problem = TesterClass(model, dataset)
+    space = problem.model_class[MODEL_PARAMETERS]
+
+    # initialization
+    estimator = base_estimator(random_state=seed_value)
+    dimensions = [v[0] for k, v in space.items()]
+    solver = Optimizer(dimensions, estimator, random_state=seed_value)
+
+    trace = []
+    best_y = np.inf
+    best_x = None
+
+    # optimization loop
+    for i in range(n_calls):
+        point_list = solver.ask()
+
+        point_dct = {k: v for k, v in zip(space.keys(), point_list)}  # convert list of dimension values to dictionary
+        objective_at_point = -problem.evaluate(point_dct) # the result of "evaluate" is accuracy / r^2, which is the more the better
+
+        if best_y > objective_at_point:
+            best_y = objective_at_point
+            best_x = point_dct
+
+        trace.append((best_x, best_y)) # remember the point, objective pair
+        print("Eval. #"+str(i))
+
+        solver.tell(point_list, objective_at_point)
+
+    return trace
+
+# the function below is necessary as a workaround for the pool.map method requiring map function to take only a single argument
+def _evaluate_optimizer(params):
+    return evaluate_optimizer(*params)
+
+def get_gaussian_surrogate(random_state):
+    return GaussianProcessRegressor(kernel=Matern(), random_state=random_state)
+
+def run(n_calls = 32,
+        n_runs = 1,
+        save_traces = True,
+        parallel_jobs = False):
+    """
+    Main function used to run the experiments.
+
+    :param n_calls: evaluation budget
+    :param n_runs: how many times to repeat the optimization in order to average out noise
+    :param save_traces: whether to save data collected during optimization
+    :param parallel_jobs: whether to run different repeats of optimization in parallel
+    :return: None
+    """
+
+    surrogates = [get_gaussian_surrogate]
```

Comment:
Just FYI, you can supply the instance directly and use the `set_params` interface to set the random_state since that seems to be your intention. (http://scikit-learn.org/stable/modules/generated/sklearn.base.BaseEstimator.html#sklearn.base.BaseEstimator.set_params)